### DISCUSSION TITLE
Has anyone managed to kill pyright yet?

### DISCUSSION THREAD

**Discussion by drcongo** [zed-industries/zed#15418] [Feature Requests]
Now that we have Ruff support, I'd really like Pyright to die in a fire. I've got this in my settings...

```
"languages": {
      "Python": {
        "format_on_save": { "language_server": { "name": "ruff" } },
        "formatter": { "language_server": { "name": "ruff" } },
        "language_servers": ["ruff"]
      }
  }
```

And I've tried it with `"language_servers": ["!pyright", "ruff"]` but still it insists on underlining things that it has no business doing so.

**Comment by drcongo** [comment]
I think this is now fixed in the preview releases. I've managed to work an entire day in Zed without Pyright sticking its oar in.

**Reply by yuxiaoy1** [reply]
Can you share your app version and ruff settings?

**Reply by drcongo** [reply]
Bit late here sorry, but I'm using the preview release channel so always on the latest preview. 

```
"languages": {
    "Python": {
      "language_servers": ["ruff", "pylsp", "python-refactoring", "!basedpyright", "!pyright"],
      "format_on_save": "on",
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        },
        {
          "language_server": {
            "name": "ruff"
          }
        }
      ]
    }
  },
  "lsp": {
    "ruff": {
      "initialization_options": {
        "settings": {
          // Ruff server settings goes here
          "lineLength": 100,
          "lint": {
            "extendSelect": ["I"]
          }
        }
      }
    },
    "pylsp": {
      "settings": {
        "plugins": {
          "rope": {
            "enabled": true
          }
        }
      }
    }
  },
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
